editor/project_manager.cpp

```diff
@@ -2473,6 +2533,91 @@ void ProjectManager::_rename_project() {
 	}
 }
 
+void ProjectManager::_manage_project_tags() {
+	for (int i = 0; i < project_tags->get_child_count(); i++) {
+		project_tags->get_child(i)->queue_free();
+	}
+
+	const ProjectList::Item &item = _project_list->get_selected_projects()[0];
+	current_project_tags = item.tags;
+	for (const String &tag : current_project_tags) {
+		ProjectTag *tag_control = memnew(ProjectTag(tag));
+		project_tags->add_child(tag_control);
+		tag_control->connect_button_to(callable_mp(this, &ProjectManager::_delete_project_tag).bind(tag));
+	}
+
+	tag_manage_dialog->popup_centered(Vector2i(500, 0) * EDSCALE);
+}
+
+void ProjectManager::_add_project_tag(const String &p_tag) {
+	if (current_project_tags.has(p_tag)) {
+		return;
+	}
+	current_project_tags.append(p_tag);
+
+	ProjectTag *tag_control = memnew(ProjectTag(p_tag));
+	project_tags->add_child(tag_control);
+	tag_control->connect_button_to(callable_mp(this, &ProjectManager::_delete_project_tag).bind(p_tag));
+}
+
+void ProjectManager::_delete_project_tag(const String &p_tag) {
+	current_project_tags.erase(p_tag);
+	for (int i = 0; i < project_tags->get_child_count(); i++) {
+		ProjectTag *tag_control = Object::cast_to<ProjectTag>(project_tags->get_child(i));
+		if (tag_control && tag_control->get_tag() == p_tag) {
+			memdelete(tag_control);
+			break;
+		}
+	}
+}
+
+void ProjectManager::_apply_project_tags() {
+	ProjectList::Item &item = _project_list->get_selected_projects().write[0];
+
+	PackedStringArray tags;
+	for (int i = 0; i < project_tags->get_child_count(); i++) {
+		ProjectTag *tag_control = Object::cast_to<ProjectTag>(project_tags->get_child(i));
+		if (tag_control) {
+			tags.append(tag_control->get_tag());
+		}
+	}
+
+	ConfigFile cfg;
+	String project_godot = item.path.path_join("project.godot");
+	Error err = cfg.load(project_godot);
+	if (err != OK) {
+		// set_message(vformat(TTR("Couldn't load project at '%s' (error %d). It may be missing or corrupted."), project_godot, err), MESSAGE_ERROR);
+	} else {
+		cfg.set_value("application", "config/tags", tags);
+		err = cfg.save(project_godot);
+		// if (err != OK) {
+		// 	set_message(vformat(TTR("Couldn't save project at '%s' (error %d)."), project_godot, err), MESSAGE_ERROR);
+		// }
+	}
+
+	_on_projects_updated();
+}
+
+void ProjectManager::_set_new_tag_name(const String p_name) {
```

Comment:
I'd add more constraints to ensure better consistency across projects:

- No dashes, only underscores (or no underscores, only dashes).
- No uppercase characters (if display is an issue, we can use `capitalize()` when displaying).

Reply:
Ok I added an array of not allowed characters. Right now it has `/` `\` and `-`, but more can be added.
Also names are now capitalized on the project list:
![image](https://user-images.githubusercontent.com/2223172/226109462-8b01786d-b8a2-4131-888e-1a3bb2b5afa2.png)